### DISCUSSION TITLE
C#: `Godot.Collections.Array.IndexOf()` and other similar methods should use the element type's implementation of `Equals()`

### DISCUSSION THREAD

**Discussion by dannflor** [godotengine/godot-proposals#7039] [Scripting]
I was trying to get `IndexOf()` to find an object so I overrode `Equals()`. While this would've worked with a native C# array, it didn't work for a Godot Array. Shouldn't it?

**Comment by AThousandShips** [comment]
* ~~This is already tracked, at least formally, in #6889~~

As `IndexOf` uses `find` and a custom function can be used there too, though currently none of the custom functions like `bsearch_custom` are available in C# it seems, changing the default behaviour isn't supported here as custom equality operators aren't supported, it will always use the builtin equality for objects

Edit: correction, see godotengine/godot#71786 for the reasoning behind that

However, the search like this won't allow custom equality at least not at the moment due to compatibility with core
